app/services/offer_validations.rb

```diff
@@ -41,4 +42,24 @@ def ratifying_provider_changed?
       errors.add(:base, :different_ratifying_provider)
     end
   end
+
+  def restrict_reverting_rejection
+    if application_choice.rejected_by_default
+      errors.add(:base, :application_rejected_by_default)
+    end
+
+    if application_choice.application_form.apply_1? && any_accepted_offers?
+      errors.add(:base, :other_offer_already_accepted)
+    end
+
+    if application_choice.application_form.apply_2? && !application_choice.self_and_siblings.last.eql?(application_choice)
```

Comment:
Is a new application form create for apply 2 or does the existing one transition to apply_2? 
Also this may require ordering to identify which is the last application choice 🤔